src/MgvOfferTaking.sol

```diff
@@ -695,12 +699,18 @@ abstract contract MgvOfferTaking is MgvHasOffers {
       }
 
       /* As an invariant, `applyPenalty` is only called when `mgvData` is not in `["mgv/notExecuted","mgv/tradeSuccess"]` */
+      // FIXME: nor if `mgvData` is in `["mgv/notEnoughGasForMakerTrade","mgv/takerTransferFail"]
```

Comment:
`mgvData` is passed from `internalMarketOrder`, where it is already not possible for `mgvData` to have the values `"mgv/notEnoughGasForMakerTrade"` or `"mgv/takerTransferFail"`. `mgvData` comes from the line 

```
(gasused, makerData, mgvData) = execute(offerList, mor, sor);
```

and `execute` can only return the following values for `mgvData`:
- `"mgv/notExecuted"`
- `"mgv/tradeSuccess"`
- `"mgv/makerRevert"`
- `"mgv/makerTransferFail"`
- `"mgv/makerReceiveFail"`

(`execute` does have its own internal `mgvData` variable that can also hold the `"mgv/notEnoughGasForMakerTrade"`,`"mgv/takerTransferFail"` values, but it reverts when that is the case).

On the other hand, if we want to cover all values possibly held by a variable called `mgvData` in this file, we should also mention that `mgvData` must be different from `"mgv/swapError"`.

Reply:
I added the comment because one must read a lot of code to infer what you just wrote :-) Seeing a comment that mentions a few possible values but not all makes me question "why aren't the other ones mentioned?" - and then I have to re-read all that code.

It sounds like you have a clear picture of the possible values for `mgvData` in different contexts. But that does not appear to be made explicit in the code? At least I've never picked up on it and have to check whenever I need it...

Not sure how to make it clearer. Except from using different names + comments that clarify what the possible values are?

Reply:
Different names is a good idea! Also just extending the list of values it cannot have + clarifications. As you see fit :)